src/rdict.rs

```diff
@@ -647,18 +638,15 @@ impl Rdict {
             None => ReadOptionsPy::default(py)?,
             Some(opt) => opt.clone(),
         };
-        if let Some(db) = &self.db {
-            Ok(RdictIter::new(
-                db,
-                &self.column_family,
-                read_opt,
-                &self.loads,
-                self.opt_py.raw_mode,
-                py,
-            )?)
-        } else {
-            Err(PyException::new_err("DB already closed"))
-        }
+
+        Ok(RdictIter::new(
+            self.get_db()?,
+            &self.column_family,
+            read_opt,
+            &self.loads,
+            self.opt_py.raw_mode,
+            py,
+        )?)
```

Comment:
Looks like we can have just
```rust
RdictIter::new(
    self.get_db()?,
    &self.column_family,
    read_opt,
    &self.loads,
    self.opt_py.raw_mode,
    py,
)
```
without `Ok(...?)`.

I wrote this, but can you change here?